src/python/pants/python/python_setup.py

```diff
@@ -50,6 +50,24 @@ def register_options(cls, register):
                 "on the format of constraint files and how constraints are applied in Pex and Pip."
             ),
         )
+        register(
+            "--use-all-requirements",
+            advanced=True,
+            # TODO: Should this default to True? Seems like a better initial experience for most
+            #  users (although they still require at least a trivial lockfile to get any benefit).
+            default=False,
+            type=bool,
+            help=(
+                "If set, all requirements will be resolved once and the result will be used by "
+                "any work that requires any subset of the requirements.  If unset, each "
```

Comment:
It's important to explain where it gets set from: the `requirement_constraints` option in `[python-setup]`.

Also, that this feature means that you can no longer have inline requirements like described by https://www.pantsbuild.org/v2.0/docs/python-third-party-dependencies.

Reply:
Well, you can as long as those are covered by your lockfile. We'll have to document that.

Reply:
What I mean by inline requirements is the ability to have decentralized requirements via `python_requirement_library`. With this PR, a requirement must be defined in the lockfile for it to actually be used anywhere. Say you have a tiny script separate from your codebase that needs `requests`; now your whole repo depends on that.

Technically, you could still have a `python_requirement_library` defined separately in a BUILD file. But we would ignore it.

In fact, if we don't want to merge, then we should consider erroring if we detect requirements that are not covered by your requirements.txt. I could see that being really confusing to a user why after following the advice of https://www.pantsbuild.org/v2.0/docs/python-third-party-dependencies#defining-inline-requirements, the req still isn't showing up. It wouldn't be obvious that they instead must update their global `requirements.txt`.

Reply:
@stuhood not sure what you meant by "factoring these options as "default lockfile"."